src/Mantasflowers.Services/Services/User/UserService.cs

```diff
@@ -0,0 +1,98 @@
+﻿using AutoMapper;
+using Mantasflowers.Contracts.User.Response;
+using Mantasflowers.Services.Repositories;
+using Mantasflowers.Services.Services.Exceptions;
+using System;
+using System.Threading.Tasks;
+
+namespace Mantasflowers.Services.Services.User
+{
+    public class UserService : IUserService
+    {
+        private readonly FirebaseService.FirebaseService _fbService;
+        private readonly IUserRepository _userRepository;
+        private readonly IMapper _mapper;
+
+        public UserService(FirebaseService.FirebaseService fbService,
+            IUserRepository userRepository,
+            IMapper mapper)
+        {
+            _userRepository = userRepository;
+            _mapper = mapper;
+            _fbService = fbService;
+        }
+
+        public async Task<GetUserResponse> GetUserByUidAsync(string uid)
+        {
+            var user = await _userRepository.GetUserByUidAsync(uid);
+            if (user == null)
+            {
+                throw new FirebaseUidNotFoundException("No user could be found for Uid");
```

Comment:
Recurring same message. Will be hard to change in the future.

Reply:
Not sure what you mean. Could you elaborate more?

Reply:
For consistency, if you'd like to, in two or more places, replace "No user could be found for Uid" with something else (or any other recurring message), this would be prove to be difficult. Also, it is a subject to change, rather than something that breaks the code on change (e.x. Roles = "admin").

Reply:
But the exception message depends on the place and context. As long as the exception type remains the same, it shouldn't mater, shouldn't it?

Also, how would Roles = "admin" break this? If changes in controllers break our service code, its a much deeper problem 🕺

Reply:
Roles = "admin" - was an example if we had an existing user base with already assigned roles. Renaming "admin" to "administrator" would break authorization. On the other hand, changing messages doesn't break anything. What I meant, is that in two or more places there are two or more identical strings. If the message is the same, then the context and place is so too.

Reply:
Ok, I think I got the point now. But I still don't see how you want to solve it. Like this?
```csharp
public async Task<GetUserResponse> GetUserByUidAsync(string uid)
{
    // ...
    if (user == null)
    {
        throw new FirebaseUidNotFoundException($"No user could be found for {nameof(uid)}");
    }
    // ...
}
```